uisdk/src/main/java/tech/dojo/pay/uisdk/domain/mapper/PaymentIntentDomainEntityMapper.kt

```diff
@@ -3,57 +3,78 @@ package tech.dojo.pay.uisdk.domain.mapper
+                    ),
+                ),
+                supportedCardsSchemes = requireNotNull(raw.merchantConfig?.supportedPaymentMethods?.cardSchemes?.mapNotNull { it }),
+                supportedWalletSchemes = raw.merchantConfig?.supportedPaymentMethods?.wallets
+                    ?: emptyList(),
+                itemLines = raw.itemLines?.map {
+                    ItemLinesDomainEntity(
+                        amount = it.amountTotal,
+                        caption = it.caption,
+                    )
+                },
+                collectionEmailRequired = raw.config?.customerEmail?.collectionRequired ?: false,
+                collectionBillingAddressRequired = raw.config?.billingAddress?.collectionRequired
+                    ?: false,
+                collectionShippingAddressRequired = raw.config?.shippingDetails?.collectionRequired
+                    ?: false,
+                isVirtualTerminalPayment = raw.paymentSource?.let { it.lowercase() == "virtual-terminal" }
+                    ?: false,
+                isPreAuthPayment = raw.captureMode?.let { it.lowercase() == "manual" } ?: false,
+                orderId = raw.reference ?: "",
+                isSetUpIntentPayment = !raw.merchantInitiatedType.isNullOrBlank() && !raw.paymentSource.isNullOrBlank(),
+                merchantName = raw.config?.tradingName ?: "",
+                isPaymentAlreadyCollected =
+                PaymentIntentStatusDomainEntity.fromStatus(requireNotNull(raw.status)) == PaymentIntentStatusDomainEntity.CAPTURED ||
+                    PaymentIntentStatusDomainEntity.fromStatus(requireNotNull(raw.status)) == PaymentIntentStatusDomainEntity.AUTHORIZED,
+            )
+        }
     }
 
-    private fun checkInvalidParameters(raw: PaymentIntentPayload) {
+    private fun checkInvalidParameters(raw: PaymentIntentPayload): Boolean {
         val invalidParams: MutableList<String> = mutableListOf()
         if (raw.id == null) invalidParams.add("id")
         if (raw.clientSessionSecret == null) invalidParams.add("clientSessionSecret")
         if (raw.amount == null && raw.intendedAmount == null) invalidParams.add("amount")
+        if (!isValidCurrencyCode(raw.amount?.currencyCode ?: raw.intendedAmount?.currencyCode)) invalidParams.add("currencyCode")
         if (raw.merchantConfig == null) invalidParams.add("merchantConfig")
         if (raw.merchantConfig?.supportedPaymentMethods == null) invalidParams.add("supportedPaymentMethods")
         if (raw.merchantConfig?.supportedPaymentMethods?.cardSchemes == null) invalidParams.add("cardSchemes")
         if (raw.status == null) invalidParams.add("status")
-        if (invalidParams.isNotEmpty()) throw EssentialParamMissingException(invalidParams)
+        return invalidParams.isNotEmpty()
+    }
+
+    private fun isValidCurrencyCode(currencyCode: String?): Boolean {
+        return currencyCode?.let {
+            try {
+                Currency.getInstance(it)
+                true
```

Comment:
Looking at the code of `Currency.getInstance(it)` it can return null, so if null I understand as an invalid right?
```suggestion
                Currency.getInstance(it) != null
```